src/isomorphic/classic/element/ReactElement.js

```diff
@@ -230,8 +230,18 @@ ReactElement.createElement = function(type, config, children) {
     props.children = childArray;
   }
 
+  var typeDisplayName = typeof type === 'function'
+    ? type.displayName || type.name || 'Unknown'
+    : type;
+
   // Resolve default props
   if (type && type.defaultProps) {
+    warning(
+      typeof type.defaultProps !== 'function',
+      'Expected `%s` in `%s` to be a class property or static getter but instead it is function.',
+      type.defaultProps.name,
+      typeDisplayName,
+    );
```

Comment:
I don't think this is the correct place to do this type of validation. We currently warn [if `defaultProps` is an instance property]( https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberClassComponent.js#L248-L255) in `ReactFiberClassComponent`. We should take a similar approach here.